ckanext/saml2/saml2_mapper.py

```diff
@@ -0,0 +1,25 @@
+role_equality = dict(
+    member='member',
+    reader='member',
+    editor='editor',
+    admin='admin'
+)
+
+
+def nsw_org_mapper(saml_info):
+    """Prepare org_dict using org=role[,role][|repeat] format."""
+    tenancy = saml_info.get('tenancy', [])
+    if tenancy:
+        org_dict = dict([
+            _get_privileged_role(*part.split('='))
+            for part in tenancy[0].split('|')
+        ])
+        return org_dict
+
+
+def _get_privileged_role(org, roles, separator=','):
+    """Return tuple with organization and most privileged role."""
+    capacity = sorted([role_equality.get(
+        role, 'member') for role in roles.split(separator)]).pop(0)
```

Comment:
If no match is found in `role_equality` it should default to no role. This is just more cautious behaviour.

Also, selection of most privileged role relies on alphabetic sort. I suggest adding a comment about this.